docs/content/docs/concepts/signals.md

```diff
@@ -6,6 +6,95 @@ discussion: 5469
+
+```typescript
+const containerSchema: ContainerSchema = {
+    initialObjects: {
+        signaler: Signaler,
+    },
+};
+
+const { container, services } = await client.createContainer(containerSchema);
+
+const signaler = container.initialObjects.signaler as Signaler;
+```
+
+`signaler` can then be directly used in your Fluid application!
+
+For more information on using `ContainerSchema` to create objects please see [Data modeling](https://fluidframework.com/docs/build/data-modeling/).
+
+## API
+`Signaler` provides a few simple methods to send signals and add/remove listeners to specific signals as well:
+- `submitSignal(signalName: string, payload?: Jsonable)` - Sends a signal with a payload to its connected listeners
+- `onSignal(signalName: string, listener: SignalListener)` - Adds a listener for the specified signal. Similar behavior as EventEmitter's `on` method.
+- `offSignal(signalName: string, listener: SignalListener)` - Removes a listener for the specified signal. Similar behavior as EventEmitter's `off` method.
+
+## Common Patterns
+### Signal Request
+When a client joins a collaboration session, they may need to receive pertinent information immediately after connecting the container.  To support this, they can request a specific signal be sent to them from other connected clients within the application. For example, in the [PresenceTracker](https://github.com/microsoft/FluidFramework/tree/main/examples/data-objects/presence-tracker) we define a "focusRequest" signal type that a newly joining client uses to request the focus-state of each currently connected client:
+
+```typescript
+private static readonly focusRequestType = "focusRequest";
+```
+
+```typescript
+container.on("connected", () => {
+    this.signaler.submitSignal(FocusTracker.focusRequestType);
+});
+```
+
+The connected clients are listening to this focus request signal, and they respond with their current focus state:
+
+```typescript
+this.signaler.onSignal(FocusTracker.focusRequestType, () => {
+    this.sendFocusSignal(document.hasFocus());
+});
+```
+When there are a lot of connected clients, usage of this request pattern can lead to high signal costs incurred from large amounts of signals being submitted all at the same time. While this pattern is helpful when a client is in need of relevant information, to limit signal costs it would be beneficial to examine whether or not the requested data will be quickly avaiable from other events being listened to within the application. The mouse tracking in [PresenceTracker](https://github.com/microsoft/FluidFramework/tree/main/examples/data-objects/presence-tracker) is an example where a newly connecting client is not required to request a signal to receive every current mouse position on the document. Since mouse movements are frequent, the newly connecting client can simply wait to recieve other users mouse positions on their mousemove events.
```

Comment:
```suggestion
This pattern adds cost however, as it forces every connected client to generate a signal.  Consider whether your scenario can be satisfied by receiving the signals naturally over time instead of requesting the information up-front. The mouse tracking in [PresenceTracker](https://github.com/microsoft/FluidFramework/tree/main/examples/data-objects/presence-tracker) is an example where a newly connecting client does not request current state. Since mouse movements are frequent, the newly connecting client can instead simply wait to receive other users' mouse positions on their next mousemove event.
```